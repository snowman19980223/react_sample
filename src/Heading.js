import styled from 'styled-components'
import PropTypes from 'prop-types'
import {TYPOGRAPHY, COMMON, get} from './constants'
import theme from './theme'

const Heading = styled.h2`
  font-weight: ${get('fontWeights.bold')};
  font-size: ${get('fontSizes.5')};
  margin: 0;
  ${TYPOGRAPHY} ${COMMON};
`

Heading.defaultProps = {
  theme
}

Heading.propTypes = {
  theme: PropTypes.object,
  ...COMMON.propTypes,
  ...TYPOGRAPHY.propTypes
}

export default Heading
