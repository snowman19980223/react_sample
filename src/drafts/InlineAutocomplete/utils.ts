import getCaretCoordinates from '@koddsson/textarea-caret'
import {Children, EventHandler, SyntheticEvent} from 'react'

import {Coordinates, ShowSuggestionsEvent, Suggestion, TextInputCompatibleChild, Trigger} from './types'

const singleWordTriggerTerminators = new Set([' ', '\n'])
const multiWordTriggerTerminators = new Set(['.', '\n'])

const isWhitespace = (char: string) => /\s/.test(char)

/**
 * Calculate whether or not suggestions should be shown based on the given state of the
 * input. If they should be shown, returns the show event.
 */
export const calculateSuggestionsQuery = (
  triggers: Array<Trigger>,
  text: string,
  caretLocation: number
): ShowSuggestionsEvent | null => {
  // Build backwards from the caret location until the most recent trigger character or terminator
  for (
    let i = caretLocation - 1, query = '', potentialTriggers = triggers;
    i >= 0 && potentialTriggers.length > 0;
    i--
  ) {
    const character = text[i]

    if (singleWordTriggerTerminators.has(character)) potentialTriggers = potentialTriggers.filter(t => t.multiWord)
    if (multiWordTriggerTerminators.has(character)) potentialTriggers = potentialTriggers.filter(t => !t.multiWord)

    for (const trigger of potentialTriggers.filter(t => character === t.triggerChar)) {
      // Trigger chars must always be preceded by whitespace or be the first character in the input,
      // and even a multi-word query cannot start with whitespace
      if ((i === 0 || isWhitespace(text[i - 1])) && !isWhitespace(query[0])) return {trigger, query}

      potentialTriggers = potentialTriggers.filter(t => t !== trigger)
    }

    query = character + query
  }

  return null
}

/**
 * Obtain the coordinates (px) of the bottom left of a character in an input, relative to the
 * top-left corner of the input itself.
 * @param input The target input element.
 * @param index The index of the character to calculate for.
 * @param adjustForScroll Control whether the returned value is adjusted based on scroll position.
 */
export const getCharacterCoordinates = (
  input: HTMLTextAreaElement | HTMLInputElement | null,
  index: number,
  adjustForScroll = true
): Coordinates => {
  if (!input) return {top: 0, left: 0}

  // word-wrap:break-word breaks the getCaretCoordinates calculations (a bug), and word-wrap has
  // no effect on input element anyway
  if (input instanceof HTMLInputElement) input.style.wordWrap = ''

  let coords = getCaretCoordinates(input, index)

  // The library calls parseInt on the computed line-height of the element, failing to account for
  // the possibility of it being 'normal' (another bug). In that case, fall back to a rough guess
  // of 1.2 based on MDN: "Desktop browsers use a default value of roughly 1.2".
  if (isNaN(coords.height)) coords.height = parseInt(getComputedStyle(input).fontSize) * 1.2

  // Sometimes top is negative, incorrectly, because of the wierd line-height calculations around
  // border-box sized single-line inputs.
  coords.top = Math.abs(coords.top)

  // For some single-line inputs, the rightmost character can be accidentally wrapped even with the
  // wordWrap fix above. If this happens, go back to the last usable index
  let adjustedIndex = index
  while (input instanceof HTMLInputElement && coords.top > coords.height) {
    coords = getCaretCoordinates(input, --adjustedIndex)
  }

  const scrollTopOffset = adjustForScroll ? -input.scrollTop : 0
  const scrollLeftOffset = adjustForScroll ? -input.scrollLeft : 0

  return {top: coords.top + coords.height + scrollTopOffset, left: coords.left + scrollLeftOffset}
}

/**
 * Obtain the coordinates of the bottom left of a character in an input relative to the top-left
 * of the page.
 * @param input The target input element.
 * @param index The index of the character to calculate for.
 */
export const getAbsoluteCharacterCoordinates = (
  input: HTMLTextAreaElement | HTMLInputElement | null,
  index: number
): Coordinates => {
  const {top: relativeTop, left: relativeLeft} = getCharacterCoordinates(input, index, true)
  const {top: viewportOffsetTop, left: viewportOffsetLeft} = input?.getBoundingClientRect() ?? {top: 0, left: 0}

  return {
    top: viewportOffsetTop + relativeTop,
    left: viewportOffsetLeft + relativeLeft
  }
}

export const getSuggestionValue = (suggestion: Suggestion): string =>
  typeof suggestion === 'string' ? suggestion : suggestion.value

export const getSuggestionKey = (suggestion: Suggestion): string =>
  typeof suggestion === 'string' ? suggestion : suggestion.key ?? suggestion.value

/**
 * Replace a section of a string.
 */
export const replaceSlice = (
  original: string,
  [startInclusive, endExclusive]: [number, number],
  replacement: string
) => {
  const before = original.substring(0, startInclusive)
  const after = original.substring(endExclusive)
  return before + replacement + after
}

/**
 * Attempts to assert that the child element is of a supported type. This can't be enforced
 * by the type system so it has to be done as a runtime check. This isn't foolproof - a
 * component that forwards a ref to a correct element but does not forward event handlers
 * will not work. But it's the best we can reasonably do.
 */
export function requireChildrenToBeInput(
  child: React.ReactElement,
  childRef: React.RefObject<HTMLElement>
): TextInputCompatibleChild {
  Children.only(child) // Assert that the child is lonely
  if (
    // There is no way to know what type the underlying child is until it mounts, so this
    // will always pass on first render before failing on the second render
    childRef.current &&
    !(childRef.current instanceof HTMLInputElement) &&
    !(childRef.current instanceof HTMLTextAreaElement)
  ) {
    throw new TypeError(
      `AutocompleteTextarea child must be a component that forwards a ref and props to an <input> or <textarea> element.`
    )
  }
  return child
}

/**
 * Combine several event handlers into one. The last handler in the list is called first
 * and no further handlers will be called if `event.preventDefault()` is called.
 */
export const augmentHandler =
  <E extends SyntheticEvent>(...handlers: Array<EventHandler<E> | undefined>) =>
  (event: E) => {
    for (const handler of [...handlers].reverse()) {
      if (!event.isDefaultPrevented()) handler?.(event)
    }
  }
