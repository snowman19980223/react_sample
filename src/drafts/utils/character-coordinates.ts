import getCaretCoordinates from '@koddsson/textarea-caret'

export type Coordinates = {
  top: number
  left: number
}

/**
 * Obtain the coordinates (px) of the bottom left of a character in an input, relative to the
 * top-left corner of the input itself.
 * @param input The target input element.
 * @param index The index of the character to calculate for.
 * @param adjustForScroll Control whether the returned value is adjusted based on scroll position.
 */
export const getCharacterCoordinates = (
  input: HTMLTextAreaElement | HTMLInputElement | null,
  index: number,
  adjustForScroll = true
): Coordinates => {
  if (!input) return {top: 0, left: 0}

  // word-wrap:break-word breaks the getCaretCoordinates calculations (a bug), and word-wrap has
  // no effect on input element anyway
  if (input instanceof HTMLInputElement) input.style.wordWrap = ''

  let coords = getCaretCoordinates(input, index)

  // The library calls parseInt on the computed line-height of the element, failing to account for
  // the possibility of it being 'normal' (another bug). In that case, fall back to a rough guess
  // of 1.2 based on MDN: "Desktop browsers use a default value of roughly 1.2".
  if (isNaN(coords.height)) coords.height = parseInt(getComputedStyle(input).fontSize) * 1.2

  // Sometimes top is negative, incorrectly, because of the wierd line-height calculations around
  // border-box sized single-line inputs.
  coords.top = Math.abs(coords.top)

  // For some single-line inputs, the rightmost character can be accidentally wrapped even with the
  // wordWrap fix above. If this happens, go back to the last usable index
  let adjustedIndex = index
  while (input instanceof HTMLInputElement && coords.top > coords.height) {
    coords = getCaretCoordinates(input, --adjustedIndex)
  }

  const scrollTopOffset = adjustForScroll ? -input.scrollTop : 0
  const scrollLeftOffset = adjustForScroll ? -input.scrollLeft : 0

  return {top: coords.top + coords.height + scrollTopOffset, left: coords.left + scrollLeftOffset}
}

/**
 * Obtain the coordinates of the bottom left of a character in an input relative to the top-left
 * of the page.
 * @param input The target input element.
 * @param index The index of the character to calculate for.
 */
export const getAbsoluteCharacterCoordinates = (
  input: HTMLTextAreaElement | HTMLInputElement | null,
  index: number
): Coordinates => {
  const {top: relativeTop, left: relativeLeft} = getCharacterCoordinates(input, index, true)
  const {top: viewportOffsetTop, left: viewportOffsetLeft} = input?.getBoundingClientRect() ?? {top: 0, left: 0}

  return {
    top: viewportOffsetTop + relativeTop,
    left: viewportOffsetLeft + relativeLeft
  }
}
