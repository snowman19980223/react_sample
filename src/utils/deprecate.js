/* eslint-disable no-console */

export default function({name, message, version}){
  if (process.env.NODE_ENV !== "production") {
    console.warn(`WARNING! ${name} is deprecated and will be removed in version ${version}. ${message}`)
  }
}
