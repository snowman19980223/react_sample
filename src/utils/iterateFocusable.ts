/**
 * Returns an iterator over all of the focusable elements within `container`.
 * Note: If `container` is itself focusable it will be included in the results.
 * @param container The container over which to find focusable elements.
 * @param reverse If true, iterate backwards through focusable elements.
 */
export function* iterateFocusableElements(
  container: Element,
  reverse = false
): Generator<HTMLElement, undefined, never> {
  const walker = document.createTreeWalker(container, NodeFilter.SHOW_ELEMENT, {
    acceptNode: node =>
      node instanceof HTMLElement && node.tabIndex === -1 ? NodeFilter.FILTER_SKIP : NodeFilter.FILTER_ACCEPT
  })
  let nextNode: Node | null = null;

  // If iterating in reverse, continue traversing down into the last child until we reach
  // a leaf DOM node
  if (reverse) {
      let lastChild = walker.lastChild()
      while (lastChild) {
        nextNode = lastChild
        lastChild = walker.lastChild()
      }
  } else {
    nextNode = walker.firstChild()
  }
  while (nextNode instanceof HTMLElement) {
    yield nextNode
    nextNode = reverse ? walker.previousNode() : walker.nextNode()
  }
  return undefined
}
