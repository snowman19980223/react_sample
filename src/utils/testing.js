import renderer from 'react-test-renderer'
import enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

enzyme.configure({adapter: new Adapter()})

/**
 * Render the component (a React.createElement() or JSX expression)
 * into its intermediate object representation with 'type',
 * 'props', and 'children' keys
 *
 * The returned object can be matched with expect().toEqual(), e.g.
 *
 * ```js
 * expect(render(<Foo />)).toEqual(render(<div foo='bar' />))
 * ```
 */
export function render(component) {
  return renderer.create(component).toJSON()
}

/**
 * Get the HTML class names rendered by the component instance
 * as an array.
 *
 * ```js
 * expect(renderClasses(<div className='a b' />))
 *   .toEqual(['a', 'b'])
 * ```
 */
export function renderClasses(component) {
  const {
    props: {className}
  } = render(component)
  return className ? className.trim().split(' ') : []
}

/**
 * console.error() is ugly af in jest; mock it with a noop,
 * and return the mock so that you can restore it:
 *
 * ```js
 * const mock = silenceConsoleError(jest)
 * // do stuff that calls console.error()
 * mock.mockRestore() // restore console.error()
 * ```
 *
 * Note: we need to pass `jest` as a reference here because
 * jest's mocking functions are bound to a module at runtime, and
 * need to be called as methods rather than pure functions
 * exported from the jest-mock module.
 */
export function silenceConsoleError(jest) {
  return jest.spyOn(console, 'error').mockImplementation(jest.fn())
}

export function mount(component) {
  return enzyme.mount(component)
}
