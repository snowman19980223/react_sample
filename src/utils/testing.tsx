import React from 'react'
import {promisify} from 'util'
import renderer from 'react-test-renderer'
import enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import {ThemeProvider} from 'styled-components'
import {default as defaultTheme} from '../theme'

// eslint-disable-next-line @typescript-eslint/no-var-requires
const readFile = promisify(require('fs').readFile)

export const COMPONENT_DISPLAY_NAME_REGEX = /^[A-Z][A-Za-z]+(\.[A-Z][A-Za-z]+)*$/

enzyme.configure({adapter: new Adapter()})

export function mount(component: React.ReactElement) {
  return enzyme.mount(component)
}

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    interface Matchers<R> {
      toImplementSystemProps: (systemProps: any) => boolean
      toImplementSxProp: () => boolean
      toImplementSxBehavior: () => boolean
      toSetDefaultTheme: () => boolean
      toSetExports: (exports: Record<string, string>) => boolean
    }
  }
}

/**
 * Render the component (a React.createElement() or JSX expression)
 * into its intermediate object representation with 'type',
 * 'props', and 'children' keys
 *
 * The returned object can be matched with expect().toEqual(), e.g.
 *
 * ```js
 * expect(render(<Foo />)).toEqual(render(<div foo='bar' />))
 * ```
 */
export function render(component: React.ReactElement) {
  return renderer.create(component).toJSON() as renderer.ReactTestRendererJSON
}

/**
 * Render the component (a React.createElement() or JSX expression)
 * using react-test-renderer and return the root node
 * ```
 */
export function renderRoot(component: React.ReactElement) {
  return renderer.create(component).root
}

/**
 * Get the HTML class names rendered by the component instance
 * as an array.
 *
 * ```js
 * expect(renderClasses(<div className='a b' />))
 *   .toEqual(['a', 'b'])
 * ```
 */
export function renderClasses(component: React.ReactElement): string {
  const {
    props: {className}
  } = render(component)
  return className ? className.trim().split(' ') : []
}

/**
 * Returns true if a node renders with a single class.
 */
export function rendersClass(node: React.ReactElement, klass: string): boolean {
  return renderClasses(node).includes(klass)
}

export function renderWithTheme(node: React.ReactElement, theme = defaultTheme): renderer.ReactTestRendererJSON {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>)
}

export function px(value: number | string): string {
  return typeof value === 'number' ? `${value}px` : value
}

export function percent(value: number | string): string {
  return typeof value === 'number' ? `${value}%` : value
}

export function renderStyles(node: React.ReactElement): any {
  const {
    props: {className}
  } = render(node)
  return getComputedStyles(className)
}

export function getComputedStyles(className: string): Record<string, string | Record<string, string>> {
  const div = document.createElement('div')
  div.className = className

  const computed = {} as any
  for (const sheet of document.styleSheets) {
    // CSSRulesLists assumes every rule is a CSSRule, not a CSSStyleRule
    for (const rule of sheet.cssRules) {
      if (rule instanceof CSSMediaRule) {
        readMedia(rule)
      } else if (rule instanceof CSSStyleRule) {
        readRule(rule, computed)
      } else {
        // console.warn('rule.type =', rule.type)
      }
    }
  }

  return computed

  function matchesSafe(node: HTMLDivElement, selector: string) {
    if (!selector) {
      return false
    }
    try {
      return node.matches(selector)
    } catch (error) {
      return false
    }
  }

  function readRule(rule: CSSStyleRule, dest: any) {
    if (matchesSafe(div, rule.selectorText)) {
      const {style} = rule
      for (let i = 0; i < style.length; i++) {
        const prop = style[i]
        dest[prop] = style.getPropertyValue(prop)
      }
    } else {
      // console.warn('no match:', rule.selectorText)
    }
  }

  function readMedia(mediaRule: CSSMediaRule) {
    const key = `@media ${mediaRule.media[0]}`
    // const dest = computed[key] || (computed[key] = {})
    const dest = {}
    for (const rule of mediaRule.cssRules) {
      if (rule instanceof CSSStyleRule) {
        readRule(rule, dest)
      }
    }

    // Don't add media rule to computed styles
    // if no styles were actually applied
    if (Object.keys(dest).length > 0) {
      computed[key] = dest
    }
  }
}
/**
 * This provides a layer of compatibility between the render() function from
 * react-test-renderer and Enzyme's mount()
 */
export function getProps(node: React.ReactElement) {
  return typeof node.props === 'function' ? node.props() : node.props
}

export function getClassName(node: React.ReactElement) {
  return getProps(node).className
}

export function getClasses(node: React.ReactElement) {
  const className = getClassName(node)
  return className ? className.trim().split(/ +/) : []
}

export async function loadCSS(path: string) {
  const css = await readFile(require.resolve(path), 'utf8')
  const style = document.createElement('style')
  style.setAttribute('data-path', path)
  style.textContent = css
  document.head.appendChild(style)
  return style
}

export function unloadCSS(path: string) {
  const style = document.querySelector(`style[data-path="${path}"]`)
  if (style) {
    style.remove()
    return true
  }
}

// If a component requires certain props or other conditions in order
// to render without errors, you can pass a `toRender` function that
// returns an element ready to be rendered.

interface Options {
  skipAs?: boolean
  skipSx?: boolean
}

interface BehavesAsComponent {
  Component: React.ComponentType<any>
  systemPropArray: any[]
  toRender?: () => React.ReactElement
  options?: Options
}

export function behavesAsComponent({Component, systemPropArray, toRender, options}: BehavesAsComponent) {
  options = options || {}

  const getElement = () => (toRender ? toRender() : <Component />)

  if (!options.skipSx) {
    it('implements sx prop behavior', () => {
      expect(getElement()).toImplementSxBehavior()
    })
  }

  if (!options.skipAs) {
    it('respects the as prop', () => {
      const As = React.forwardRef<HTMLDivElement>((_props, ref) => <div className="as-component" ref={ref} />)
      const elem = React.cloneElement(getElement(), {as: As})
      expect(render(elem)).toEqual(render(<As />))
    })
  }

  it('sets a valid displayName', () => {
    expect(Component.displayName).toMatch(COMPONENT_DISPLAY_NAME_REGEX)
  })

  it('sets the default theme', () => {
    expect(getElement()).toSetDefaultTheme()
  })

  it('renders consistently', () => {
    expect(render(getElement())).toMatchSnapshot()
  })
}

export function checkExports(path: string, exports: Record<any, any>): void {
  it('has declared exports', () => {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const mod = require(`../${path}`)
    expect(mod).toSetExports(exports)
  })
}
