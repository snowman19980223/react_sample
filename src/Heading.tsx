import styled from 'styled-components'
import {COMMON, get, SystemCommonProps, SystemTypographyProps, TYPOGRAPHY} from './constants'
import sx, {SxProp} from './sx'
import theme from './theme'
import {ComponentProps} from './utils/types'

const Heading = styled.h2<SystemTypographyProps & SystemCommonProps & SxProp>`
  font-weight: ${get('fontWeights.bold')};
  font-size: ${get('fontSizes.5')};
  margin: 0;
  ${TYPOGRAPHY};
  ${COMMON};
  ${sx};
`

Heading.defaultProps = {
  theme
}

export type HeadingProps = ComponentProps<typeof Heading>
export default Heading
