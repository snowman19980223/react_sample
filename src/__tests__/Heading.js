/* eslint-disable jsx-a11y/heading-has-content */
import React from 'react'
import Heading from '../Heading'
import {render} from '../utils/testing'

const theme = {
  breakpoints: ['400px', '640px', '960px', '1280px'],
  colors: {
    green: ['#010', '#020', '#030', '#040', '#050', '#060'],
  },
  fontSizes: [12, 14, 16, 20, 24, 32, 40, 48],
  fonts: {
    normal: 'Helvetica,sans-serif',
    mono: 'Consolas,monospace'
  },
  lineHeights: {
    normal: 1.5,
    condensed: 1.25,
    condensedUltra: 1
  }
}

describe('Heading', () => {
  it('renders <h1> by default', () => {
    expect(render(<Heading />).type).toEqual('h1')
  })

  it('respects the is prop', () => {
    expect(render(<Heading is="h6" />).type).toEqual('h6')
  })
  it('renders margin', () => {
    expect(render(<Heading m={1} theme={theme} />)).toHaveStyleRule('margin', '4px')
    expect(render(<Heading m={[0, 1, 2, 3, 4]} theme={theme} />)).toMatchSnapshot()
    expect(render(<Heading m={[1, 1, 3, 3]} theme={theme} />)).toMatchSnapshot()
  })

  it('renders padding', () => {
    expect(render(<Heading p={1} theme={theme} />)).toHaveStyleRule('padding', '4px')
    expect(render(<Heading p={[0, 1, 2, 3, 4]} theme={theme} />)).toMatchSnapshot()
    expect(render(<Heading p={[1, 1, 3, 3]} theme={theme} />)).toMatchSnapshot()
  })

  it('respects color', () => {
    expect(render(<Heading color="green.5" theme={theme} />)).toHaveStyleRule('color', theme.colors.green[5])
  })

  it('respects fontWeight', () => {
    expect(render(<Heading fontWeight="bold" theme={theme} />)).toHaveStyleRule('font-weight', 'bold')
    expect(render(<Heading fontWeight="normal" theme={theme} />)).toHaveStyleRule('font-weight', 'normal')
  })

  it('respects lineHeight', () => {
    expect(render(<Heading lineHeight="normal" theme={theme} />)).toHaveStyleRule('line-height', String(theme.lineHeights.normal))
    expect(render(<Heading lineHeight="condensed" theme={theme} />)).toHaveStyleRule('line-height', String(theme.lineHeights.condensed))
    expect(render(<Heading lineHeight="condensedUltra" theme={theme} />)).toHaveStyleRule('line-height', String(theme.lineHeights.condensedUltra))
  })

  it('respects fontFamily="mono"', () => {
    expect(render(<Heading fontFamily="mono" theme={theme} />)).toHaveStyleRule('font-family', theme.fonts.mono)
  })

  xit('respects nowrap', () => {
    expect(render(<Heading nowrap theme={theme} />)).toEqual(render(<span className="no-wrap" />))
  })

  xit('renders fontSize with f* classes using inverse scale', () => {
    expect(render(<Heading fontSize={0} theme={theme} />)).toEqual(render(<span className="f6" />))
    expect(render(<Heading fontSize={1} theme={theme} />)).toEqual(render(<span className="f5" />))
    expect(render(<Heading fontSize={2} theme={theme} />)).toEqual(render(<span className="f4" />))
    expect(render(<Heading fontSize={3} theme={theme} />)).toEqual(render(<span className="f3" />))
    expect(render(<Heading fontSize={4} theme={theme} />)).toEqual(render(<span className="f2" />))
    expect(render(<Heading fontSize={5} theme={theme} />)).toEqual(render(<span className="f1" />))
    expect(render(<Heading fontSize={6} theme={theme} />)).toEqual(render(<span className="f0" />))
  })

  xit('does not pass on arbitrary attributes', () => {
    const defaultOutput = render(<Heading />)
    expect(render(<Heading bugs="bar" />)).toEqual(defaultOutput)
    expect(render(<Heading hidden />)).toEqual(defaultOutput)
  })

  xit('respects other values for fontSize', () => {
    expect(render(<Heading fontSize="2em" theme={theme} />)).toHaveStyleRule('font-size', '2em')
    expect(render(<Heading fontSize={100} theme={theme} />)).toHaveStyleRule('font-size', '100px')
  })
})
