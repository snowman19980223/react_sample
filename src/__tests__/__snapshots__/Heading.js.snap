// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders margin 1`] = `
.emotion-0 {
  font-size: 32px;
  margin: 0px;
}

@media screen and (min-width:400px) {
  .emotion-0 {
    margin: 4px;
  }
}

@media screen and (min-width:640px) {
  .emotion-0 {
    margin: 8px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:1280px) {
  .emotion-0 {
    margin: 32px;
  }
}

<h1
  className="emotion-0"
/>
`;

exports[`Heading renders margin 2`] = `
.emotion-0 {
  font-size: 32px;
  margin: 4px;
}

@media screen and (min-width:400px) {
  .emotion-0 {
    margin: 4px;
  }
}

@media screen and (min-width:640px) {
  .emotion-0 {
    margin: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 16px;
  }
}

<h1
  className="emotion-0"
/>
`;

exports[`Heading renders padding 1`] = `
.emotion-0 {
  font-size: 32px;
  margin: 0px;
  padding: 0px;
}

@media screen and (min-width:400px) {
  .emotion-0 {
    padding: 4px;
  }
}

@media screen and (min-width:640px) {
  .emotion-0 {
    padding: 8px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media screen and (min-width:1280px) {
  .emotion-0 {
    padding: 32px;
  }
}

<h1
  className="emotion-0"
/>
`;

exports[`Heading renders padding 2`] = `
.emotion-0 {
  font-size: 32px;
  margin: 0px;
  padding: 4px;
}

@media screen and (min-width:400px) {
  .emotion-0 {
    padding: 4px;
  }
}

@media screen and (min-width:640px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    padding: 16px;
  }
}

<h1
  className="emotion-0"
/>
`;
